repository-meta-analyzer/src/main/java/org/hyades/processor/IntegrityAnalyzerProcessor.java

```diff
@@ -0,0 +1,301 @@
+                        .setInternal(integrityModel.getComponent().isInternal())
+                        .setMd5Hash(integrityModel.getComponent().getMd5())
+                        .setSha1Hash(integrityModel.getComponent().getSha1())
+                        .setSha256Hash(integrityModel.getComponent().getSha256())
+                        .setUuid(integrityModel.getComponent().getUuid().toString())
+                        .setComponentId(integrityModel.getComponent().getId()).build();
+
+                final IntegrityResult.Builder resultBuilder = IntegrityResult.newBuilder()
+                        .setMd5HashMatch(integrityModel.isMd5HashMatched())
+                        .setComponent(component1)
+                        .setSha1HashMatch(integrityModel.isSha1HashMatched())
+                        .setSha256HashMatch(integrityModel.isSha256HashMatched())
+                        .setRepository(repository.getUrl())
+                        .setPublished(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()));
+                return resultBuilder.build();
+            } catch (Exception ex) {
+                LOGGER.error("Failed to analyze {} using {} with repository {}",
+                        component.getPurl(), analyzer.getName(), repository.getIdentifier(), ex);
+                return null;
+            }
+
+        } else {
+            LOGGER.debug("Cache miss for integrity check (analyzer: {}, purl: {}, repository: {})", analyzer.getName(), repository.getUrl(), component.getUuid());
+            analyzer.setRepositoryBaseUrl(repository.getUrl());
+            if (Boolean.TRUE.equals(repository.isAuthenticationRequired())) {
+                try {
+                    analyzer.setRepositoryUsernameAndPassword(repository.getUsername(), secretDecryptor.decryptAsString(repository.getPassword()));
+                } catch (Exception e) {
+                    LOGGER.error("Failed decrypting password for repository: " + repository.getIdentifier(), e);
+                }
+            }
+
+            LOGGER.debug("Performing integrity check on component: {}", component.getPurl());
+            IntegrityModel integrityModel = new IntegrityModel();
+            try {
+                final var analyzerComponent = new org.hyades.persistence.model.Component();
+                analyzerComponent.setPurl(component.getPurl());
+                analyzerComponent.setMd5(component.getMd5Hash());
+                analyzerComponent.setSha1(component.getSha1Hash());
+                analyzerComponent.setSha256(component.getSha256Hash());
+                UUID uuid = UUID.fromString(component.getUuid());
+                analyzerComponent.setUuid(uuid);
+                analyzerComponent.setId((long) component.getComponentId());
+                try (CloseableHttpResponse response = analyzer.getIntegrityCheckResponse(new PackageURL(component.getPurl()))) {
```

Comment:
`getIntegrityCheckResponse` can return `null` in some cases. Should we handle that case explicitly, and do we really want to cache `null`?